base_layer/core/src/consensus/consensus_encoding/hashing.rs

```diff
@@ -41,8 +42,9 @@ where D: Default
 {
     #[allow(clippy::new_ret_no_self)]
     pub fn new(label: &'static str) -> ConsensusHasher<D> {
+        let network = *CURRENT_NETWORK.lock().unwrap();
```

Comment:
One thought I had was to make `CURRENT_NETWORK` an `Option<Network>` with `None` as the default. We can check to validate a `CURRENT_NETWORK` has been set, and if not fail to provide the hasher. It's a harder fault but would prevent the possibility of using the wrong network in the hash.

Reply:
The only real danger comes with wallets, where you think you might be on stagenet, but actually mainnet. but when you spend to other wallets, the addresses are different so this should be blocked. 

We should either way clearly print out the current selected network on base nodes, console wallets already have a network label right bottom